### PR TITLE
Add tutorial - generate data for training embeddings and reranking models 

### LINKED ISSUE #890: [DOCS] tutorial - generate data for training embeddings and reranking models
## Which page or section is this issue related to?

An addaptation of https://docs.zenml.io/user-guide/llmops-guide/finetuning-embeddings/finetuning-embeddings-with-sentence-transformers

## What are you documenting, or what change are you making in the documentation?

NA

### PR DESCRIPTION
**Pull request by davidberenstein1957**
Closes #890 

image

image

### CODE REVIEW

File: docs/sections/pipeline_samples/tutorials/GenerateSentencePair.ipynb [hunk 1]
```diff
@@ -0,0 +1,438 @@
+{
```
Comment (gabrielmbmb) [inline docs/sections/pipeline_samples/tutorials/GenerateSentencePair.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    llm = OpenAILLM(model="gpt-4o", api_key=os.getenv("OPENAI_API_KEY"))
</pre><p>Could we use <code>InferenceEndpointsLLM</code> instead? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/argilla-io/distilabel/pull/893/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

**Comment by plaguss** [comment]
It may be nice to check the `GenerateSentencePair` with `hard_negative=True`, as that option didn't exist back when I did the blog for the SDK, to see how much it improves.
Also, to upload the data to argilla, this notebook has all the blocks: https://github.com/argilla-io/argilla-sdk-chatbot/blob/main/argilla_datasets.ipynb (and the argilla space with the datasets from the blog: https://huggingface.co/spaces/plaguss/argilla-sdk-chatbot)

File: docs/sections/pipeline_samples/tutorials/GenerateSentencePair.ipynb (hunk 1 above)
Comment (davidberenstein1957) [reply docs/sections/pipeline_samples/tutorials/GenerateSentencePair.ipynb:1]: image

File: docs/sections/pipeline_samples/tutorials/GenerateSentencePair.ipynb [hunk 2]
```diff
@@ -0,0 +1,715 @@
...
```
Comment (sdiazlor) [inline docs/sections/pipeline_samples/tutorials/GenerateSentencePair.ipynb:1]: <p>We could add a !!! note at the end of the section just to indicate the components as we had mentioned some time ago: <a href="https://haystack.deepset.ai/tutorials/27_first_rag_pipeline" target="_blank">https://haystack.deepset.ai/tutorials/27_first_rag_pipeline</a> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/argilla-io/distilabel/pull/893/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (sdiazlor) [inline docs/sections/pipeline_samples/tutorials/GenerateSentencePair.ipynb:1]: <p>Maybe we can start adding the versions for third-party libraries for easier maintainability.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/argilla-io/distilabel/pull/893/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (sdiazlor) [inline docs/sections/pipeline_samples/tutorials/GenerateSentencePair.ipynb:1]: <p>Could you remove this type of outputs?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/argilla-io/distilabel/pull/893/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (sdiazlor) [inline docs/sections/pipeline_samples/tutorials/GenerateSentencePair.ipynb:1]: <p>(Optional) Deploy Argilla</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/argilla-io/distilabel/pull/893/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (sdiazlor) [inline docs/sections/pipeline_samples/tutorials/GenerateSentencePair.ipynb:1]: <p>Even if in markdown, better show it using <code></code>`python</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/argilla-io/distilabel/pull/893/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Comment by sdiazlor** [comment]
@davidberenstein1957 In general, it looks nice! Just some small comments

**Review by sdiazlor — APPROVED** [review]

**Pull request merged**